boot/bootstate20.go

```diff
@@ -219,6 +222,38 @@ func (u20 *bootStateUpdate20) commit() error {
 		expectReseal = resealExpectedByModeenvChange(u20.writeModeenv, u20.modeenv)
 	}
 
+	autoRepair, err := IsUnlockedWithRecoveryKey()
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+	} else if autoRepair {
```

Comment:
shouldn't we try to do this only if we come from MarkBootSuccessful? it seems that if we don't check for that we will reaseal/repair multiple times until we reboot if people do other operations?